### ISSUE TITLE
Add Blender Game Engine (UPBGE)

### ISSUE THREAD

**Issue by UnidayStudio** [SteamDatabase/FileDetectionRuleSets#56]
### Some reference to the engine, such as the developer site or a wikipedia article.

Link: [UPBGE Website](https://upbge.org/)

UPBGE is an open-source, 3D game engine forked from the old **Blender Game Engine** and deployed with Blender itself. This unified workflow is its main strength, as you can make your game from start to finish without leaving Blender.

### SteamDB links

https://steamdb.info/app/814590/
https://steamdb.info/app/324240/
https://steamdb.info/app/499460/
https://steamdb.info/app/788890/
https://steamdb.info/app/1324560/
https://steamdb.info/app/1375640/

### Possible way to detect it

It's very simple to detect blender game engine (upbge) games: The way the engine works is by running a player, called **"blenderplayer.exe"**. Most of the games will have this player located somewhere (99,9% of the time is in the root folder where you launch the game. 

Some rare projects (like I will escape, in the SteamDB link) will rename or "encapsulate" this player into other files (it's not the engine default, btw), but when they do this, the source files are easily recognizable by the extension **".blend"** (blender files).

Another very obvious hint (that may be a bit hard to track) is that there is a folder names by the version of the blender used to make the game in the root directory (with the blenderplayer):

- /2.72/
- ...
- /2.78/
- /2.79/
- /3.0/ (new Blender/UPBGE)

It may be a bit hard to track this folder name because it's dependent on the engine's version. But inside the folder we'll have at least a /python/ and a /datafiles/ subfolder. We may have more, but those two are guaranteed to be in there.

Some developers may also include the debug **blender.exp, blender.lib, blenderplayer.exp, blenderplayer.lib** files. 

**TLDR:**
If there is a file called "blenderplayer" (or upbgeplayer, they'll rename it to this soon), it's a BGE/UPBGE game 100%. If not but the're is a folder containing some version number (like 2.79 or 3.0) and some .blend files, it's a BGE/UPBGE game.

**Comment by Lifeismana** [comment]
Looking a bit into the depot every game have this lib `BlendThumb64.dll` do you know if it's UPBGE specific ?

**Comment by UnidayStudio** [comment]
Nice catch. I've checked here (I have plenity of different version installed) and most of them have the `BlendThumb64.dll` or a `BlendThumb.dll`. I think it's safe to look for one of those dlls or the `blenderplayer.exe`.
